wgsl/index.bs

```diff
@@ -4070,10 +4069,121 @@ Issue: in Vulkan, builtin variables occoupy I/O location slots counting toward l
 
 #### User Data Attribute TODO #### {#user-data-attributes}
 
-TODO: User data attributes must not be of [=bool=] type or contain a [=bool=] type.
+User-defined data can be passed as input to the start of a pipeline, passed
+between stages of a pipeline or output from the end of a pipeline.
+User-defined IO must not be passed to [=compute=] shader entry points.
+User-defined IO must be an [=IO-shareable=] type, but must not be or contain a
+[=bool=] or [=array=] type.
+All user defined IO must be assigned a location (See [[#input-output-locations]]).
 
-#### Input-output Locations TODO #### {#input-output-locations}
-TODO: *Stub*. Location-sizing of types, non-overlap among variables referenced within an entry point static call tree.
+#### Input-output Locations #### {#input-output-locations}
+
+Each location represents 16-bytes of storage.
```

Comment:
should we say ("up to 16")?

Reply:
I have a note saying the number of locations is defined by the API side. There is an issue in the builtin section that says Vulkan shares locations with builtins in the API.

Reply:
this isn't about number of locations, this is about the size of each

Reply:
The data item does not have to cover the whole location. 

Since no two items can share the same location (since component is not supported), perhaps:

> Each IO location can store a value of up to 16 bytes in size.

or:

> Each IO location is a unit of up to 16 bytes of storage.